samza-core/src/main/java/org/apache/samza/coordinator/CoordinationService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.coordinator;
+
+import org.apache.samza.coordinator.leaderelection.LeaderElector;
+
+
+public interface CoordinationService {
+  // life cycle
+  void start();
```

Comment:
I agree that stop/start need is not determined yet. So I will remove it for now.
In general I agree that these interfaces will need to be designed better. So I will open a jira/SEP to discuss the design. I think we should still keep them as a single CoordinationService/CoordinationUtils interface because they all are configured together, and they all need to be provided if we use a different coordination service.

I'd suggest to commit this as is and make the changes after the discussion (we will have better data on use cases).